src/main/java/step4/domain/LottoSeller.java

```diff
@@ -0,0 +1,41 @@
+package step4.domain;
+
+import step4.domain.generator.LottoGenerator;
+import step4.domain.generator.StringLottoGenerator;
+
+import java.util.stream.LongStream;
+
+import static java.util.stream.Collectors.collectingAndThen;
+import static java.util.stream.Collectors.toList;
+
+public class LottoSeller {
+
+    private final LottoGenerator randomGenerator;
+
+    public LottoSeller(LottoGenerator randomGenerator) {
+        this.randomGenerator = randomGenerator;
+    }
+
+    public Lottos buy(LottoPaper lottoPaper) {
+        Lottos manualLottos = manualLottos(lottoPaper);
+
+        Lottos autoLottos = autoLottos(lottoPaper);
+
+        return manualLottos.addLottos(autoLottos);
+    }
+
+    private Lottos autoLottos(LottoPaper lottoPaper) {
+        Lottos autoLottos = LongStream.range(0, lottoPaper.getAutoCount())
+                .mapToObj(count -> randomGenerator.generate())
+                .collect(collectingAndThen(toList(), Lottos::of));
+        return autoLottos;
```

Comment:
```suggestion
        return LongStream.range(0, lottoPaper.getAutoCount())
                .mapToObj(count -> randomGenerator.generate())
                .collect(collectingAndThen(toList(), Lottos::of));
```
변수를 다른 곳에서 사용하지 않는 다면 바로 반환 해주는 것도 좋아요 :)